tests/src/rules/default.js

```diff
@@ -1,305 +1,229 @@
-          },
         }),
         test({
           code: `import Foo from "./typescript-export-react-test-renderer"`,
-          parser,
           settings: {
-            'import/parsers': { [parser]: ['.ts'] },
             'import/resolver': { 'eslint-import-resolver-typescript': true },
           },
-          parserOptions: {
-            tsconfigRootDir: path.resolve(__dirname, '../../files/typescript-export-react-test-renderer/'),
-          },
         }),
         test({
           code: `import Foo from "./typescript-extended-config"`,
-          parser,
           settings: {
-            'import/parsers': { [parser]: ['.ts'] },
             'import/resolver': { 'eslint-import-resolver-typescript': true },
           },
-          parserOptions: {
-            tsconfigRootDir: path.resolve(__dirname, '../../files/typescript-extended-config/'),
-          },
         }),
         test({
           code: `import foobar from "./typescript-export-assign-property"`,
-          parser,
           settings: {
-            'import/parsers': { [parser]: ['.ts'] },
             'import/resolver': { 'eslint-import-resolver-typescript': true },
           },
         }),
-      ),
 
-      invalid: [
-        test({
+    // es2022: Arbitrary module namespace identifier names
+    testVersion('>= 8.7', () => ({
+      code: 'export { "default" as bar } from "./bar"',
+      parserOptions: {
+        ecmaVersion: 2022,
+      },
+    })),
+
+    ...SYNTAX_CASES,
+   ]),
+
+  invalid: parsers.all([
+    test({
+      code: "import Foo from './jsx/FooES7.js';",
+      errors: ["Parse errors in imported module './jsx/FooES7.js': Unexpected token = (6:16)"],
+    }),
+
+    test({
+      code: 'import baz from "./named-exports";',
+      errors: [{ message: 'No default export found in imported module "./named-exports".',
+        type: 'ImportDefaultSpecifier' }] }),
+
+    // es7 export syntax
+    test({
+      code: 'export baz from "./named-exports"',
+      errors: ['No default export found in imported module "./named-exports".'],
+    }),
+    test({
+      code: 'export baz, { bar } from "./named-exports"',
+      errors: ['No default export found in imported module "./named-exports".'],
+    }),
+    test({
+      code: 'export baz, * as names from "./named-exports"',
+      errors: ['No default export found in imported module "./named-exports".'],
+    }),
+    // exports default from a module with no default
+    test({
+      code: 'import twofer from "./broken-trampoline"',
+      errors: ['No default export found in imported module "./broken-trampoline".'],
+    }),
+
+    // #328: * exports do not include default
+    test({
+      code: 'import barDefault from "./re-export"',
+      errors: ['No default export found in imported module "./re-export".'],
+    }),
+
+
+    //typescript context
+    //context('TypeScript', function () 
+    test({
           code: `import foobar from "./typescript"`,
-          parser,
+          
```

Comment:
```suggestion
          features: ['ts'],
```
and same for all the TS cases